lib/integrations/awsoidc/deploydatabaseservice.go

```diff
@@ -65,6 +65,10 @@ type DeployDatabaseServiceRequest struct {
 	// ResourceCreationTags is used to add tags when creating resources in AWS.
 	ResourceCreationTags AWSTags
 
+	// DeployServiceConfigString creates a teleport.yaml configuration that the agent
+	// deployed in a ECS Cluster (using Fargate) will use.
+	DeployServiceConfigString func(proxyHostPort, iamToken string, resourceMatcherLabels types.Labels) (string, error)
```

Comment:
Looks like because we have `deployserviceconfig` package so we came to this name but it reads weird to have a name that ends with "string" but its type is of a `function`. More so when we check for its value below:
```
if r.DeployServiceConfigString == nil {
	return trace.BadParameter("deploy service config is required")
}
```
Suggestion: maybe `CreateDeployServiceConfig`?